EventProc/src/OverlayProducer.cxx

```diff
@@ -0,0 +1,329 @@
+      }
+
+
+	  std::vector <SimCalorimeterHit> outSimCaloHitColls;
+	  std::vector <SimTrackerHit> outSimTrackHitColls;
+
+	  // TO DO: get this list of collections by listing what's there in the overlay file
+	  // could we use drop/keep rules to skip anything we don't want? 
+	  //   -- or would that have to happen at overlay file production time, since Process doesn't own this file?
+	  // we *could* also use a listing from the sim event we're interested in.
+	  //   -- in many ways easier? but worried about corner cases where a collection happens to be empty for a given event. 
+
+	  
+	  // get the collections that we want to overlay, by looping over the list of collections passed to the producer : collections_
+	  for (uint iColl = 0; iColl < collections_.size(); iColl++) {
+		  
+		std::vector<SimCalorimeterHit> simHits = event.getCollection<SimCalorimeterHit>( collections_[iColl] );
+
+		// here: check if it exists -- if it's a simcalo hit collection. then define the out coll:
+
+		std::vector<SimCalorimeterHit> outHits;
+
+		bool isEcalHitCollection = false ;
+
+		if (strstr (collections_[iColl].c_str(), "Ecal") )
+		  isEcalHitCollection = true ;
+
+		std::vector<SimCalorimeterHit> overlayHits = overlayEvent->getCollection<SimCalorimeterHit>( collections_[iColl] );
+		//		std::vector<SimTrackerHit> overlayTrackerHits = overlayEvent->getCollection<SimTrackerHit>( collections_[iColl] );
+		  //dynamic_cast<SimCalorimeterHit> (overlayEvent->getCollection<SimCalorimeterHit>( collections_[iColl] )  );
+
+
+		//do something else if we're getting tracker hits. this type of check, for one, doesn't work, and simply leads to segfault if it's a simtracker hit
+		/*
+		if ( dynamic_cast<std::vector<SimCalorimeterHit>*>( &overlayHits  ) == NULL ) {
+		  
+		   ldmx_log(debug) <<  "Nope, no simcalorimeter hit collection called " << collections_[iColl] << ", skipping"  ;
+		  continue;
+		}
+		else
+		   ldmx_log(debug) <<  "It's simcalo alright" ;
+		overlayHits[0].Print();
+
+		*/
+		//if we alredy added at least one overlay event, this collection already exists 
+		if ( v_outHits.size() > iColl  ) {
+		  outHits = v_outHits[iColl];
+		} else { //otherwise, start out by just copying the sim hits, unaltered. 
+
+		  outHits=simHits;
+		  outCollNames.push_back( collections_[iColl]+"Overlay" );
+		  
+		  if (verbosity_ > 2) {
+			 ldmx_log(debug) <<  "[ OverlayProducer ]: in loop: start of collection " << outCollNames.back()  ;
+			 ldmx_log(debug) <<  "[ OverlayProducer ]: in loop: printing current sim event: "  ;
+		  }
+
+		  //we don't need to touch the hard process sim hits, really... but we might need the simhits in the hit map.
+		  for (const SimCalorimeterHit &simHit : simHits ) {
+			
+			simHit.Print();
+
+			//store ids of existing hits. for these, need to add contribs...
+			if (isEcalHitCollection) {
+			  int simHitID = simHit.getID();
+			  hitMap[ simHitID ] = SimCalorimeterHit();
```

Comment:
Instead of using the default constructor, you could just copy over the `simHit` object.
```
hitmap[simHitID] = simHit
```
Then you wouldn't have to do the `set` commands after this.